src/Orleans.Core/CodeGeneration/TypeUtils.cs

```diff
@@ -403,7 +283,7 @@ public static bool IsGrainClass(Type type)
             if (!grainType.IsAssignableFrom(type)) return false;
 
             // exclude generated classes.
-            return !IsGeneratedType(type);
+            return !type.IsDefined(typeof(GeneratedCodeAttribute), true);
```

Comment:
Shouldn't this be `inherit: false` instead?

Reply:
This was an automated inlining, but I agree